doc/reference/configuration/configuration_reference.rst

```diff
@@ -221,6 +488,718 @@ This section describes options related to :ref:`storing configuration in etcd <c
+    ``database.replicaset_uuid`` can be used to specify a replica set identifier manually.
+
+    See also: :ref:`database.instance_uuid <configuration_reference_database_instance_uuid>`
+
+    | Type: string
+    | Default: box.NULL
+    | Environment variable: TT_DATABASE_REPLICASET_UUID
+
+
+.. _configuration_reference_database_txn_isolation:
+
+.. confval:: database.txn_isolation
+
+    A transaction :ref:`isolation level <transaction_model_levels>`.
+
+    | Type: string
+    | Default: ``best-effort``
+    | Possible values: ``best-effort``, ``read-committed``, ``read-confirmed``
+    | Environment variable: TT_DATABASE_TXN_ISOLATION
+
+
+.. _configuration_reference_database_txn_timeout:
+
+.. confval:: database.txn_timeout
+
+    A timeout (in seconds) after which the transaction is rolled back.
+
+    See also: :ref:`box.begin() <box-begin>`
+
+    | Type: number
+    | Default: 3153600000
+    | Environment variable: TT_DATABASE_TXN_TIMEOUT
+
+
+.. _configuration_reference_database_use_mvcc_engine:
+
+.. confval:: database.use_mvcc_engine
+
+    Whether the :ref:`transactional manager <txn_mode_transaction-manager>` is enabled.
+
+    | Type: boolean
+    | Default: false
+    | Environment variable: TT_DATABASE_USE_MVCC_ENGINE
+
+
+
+
+
+..  _configuration_reference_iproto:
+
+iproto
+------
+
+The ``iproto`` section is used to configure parameters related to communicating to and between cluster instances.
+
+.. NOTE::
+
+    ``iproto`` can be defined in any :ref:`scope <configuration_scopes>`.
+
+-   :ref:`iproto.advertise.* <configuration_reference_iproto_advertise>`
+
+    -   :ref:`iproto.advertise.client <configuration_reference_iproto_advertise_client>`
+    -   :ref:`iproto.advertise.peer <configuration_reference_iproto_advertise_peer>`
+    -   :ref:`iproto.advertise.sharding <configuration_reference_iproto_advertise_sharding>`
+
+-   :ref:`iproto.listen <configuration_reference_iproto_listen>`
+-   :ref:`iproto.net_msg_max <configuration_reference_iproto_net_msg_max>`
+-   :ref:`iproto.readahead <configuration_reference_iproto_readahead>`
+-   :ref:`iproto.threads <configuration_reference_iproto_threads>`
+
+
+..  _configuration_reference_iproto_advertise:
+
+iproto.advertise.*
+~~~~~~~~~~~~~~~~~~
+
+The ``iproto.advertise.*`` configuration options specify URIs used to advertise the current instance to other cluster members or clients.
+These options accept an URI in the following formats:
+
+-   User :ref:`credentials <configuration_reference_credentials>` and an address: ``username@host:port`` or ``username:password@host:port``.
+
+-   User credentials: ``username@`` or ``username:password@``.
+    In this case, an advertise address is taken from :ref:`iproto.listen <configuration_reference_iproto_listen>`.
+
+-   An address: ``host:port``.
+
+If ``password`` is missing, it is taken from :ref:`credentials <configuration_reference_credentials>` for the specified ``username``.
```

Comment:
It is applicable to `peer` and `sharding`, but not `client`. The latter only accepts `host:port` (or a unix socket without credentials). See the [description](https://github.com/tarantool/tarantool/blob/2c422a0b98675cab480ebb44256dfeb72bd2f820/src/box/lua/config/instance_config.lua#L676-L723) of the options.

Reply:
It is now described in the following way:

* `iproto.advertise.*` -- all the three syntax variants.
* `iproto.advertise.client` -- has constraints, just one syntax variant.

I would prefer something like the following:

* `iproto.advertise.client` -- host:port or unix socket, no user/pass at all.
* `iproto.advertise.{peer,sharding}` -- three syntax variants.

In my head the `client` URI and the `peer`/`sharding` ones are quite different entities.

Moreover, I can just look at the `iproto.advertise.*` description and think that it is applicable to `iproto.advertise.client` without additional syntax constraints.

Reply:
Thanks for your suggestions. To me,`iproto.advertise.{peer,sharding}` looks like a bit of overengineering for docs :) So, I decided to remove the top-level `iproto.advertise.*` and described an URI format for `iproto.advertise.client` and `iproto.advertise.peer` separately. `iproto.advertise.sharding` description just tells that it has the same format as `iproto.advertise.peer. One move variant is including the content describing the URI format from `iproto.advertise.peer` to `iproto.advertise.sharding` as is, so a reader doesn't need to jump between descriptions because they will be duplicated.

> If there is a new `iproto.advertise.<some_new_param>` option is added with the same URI format, we can consider extracting the URI format description to a separate section.

Reply:
Thanks! The current variant seems nice.